### PR TITLE
Simple/minimal support for GroundOverlay layer - take no. 2

### PR DESCRIPTION
**Pull request by Simek**
Documentation:
https://developers.google.com/maps/documentation/javascript/reference#GroundOverlay

Example provided based on:
https://developers.google.com/maps/documentation/javascript/groundoverlays

PR rewritten from the scratch to match 8.x changes.

I'm only not sure if I implemented required props (url, bounds) in proper way because no other GM component in this library shares that approach.

## Screenshots

screenshot-2017-9-21 groundoverlay react google maps style guide

### CODE REVIEW

**Comment by tomchentw** [comment]
Hey @Simek , thanks for taking this! But could you take a look at some components under `src/macros` and put the new GroundOverlay there? Thus, we can auto-generate it from the Google Maps API docs with `build:src` script.

**Comment by Simek** [comment]
Hey @tomchentw, should I generate src file using given command or just leave macro file in PR?

**Comment by tomchentw** [comment]
@Simek should just leave the macro file in the PR. No `src/components`, no `lib`

**Comment by Simek** [comment]
Travis probably fails because build/regenerate command is missing in `before_script` list.

**Comment by tomchentw** [comment]
Merged #627 , please rebase.

**Comment by Simek** [comment]
Seems that for the first run `npm run build:src` wasn't included in Travis build. I don't know if you retriggered it manually but in second run line was present and CI passed.

Is there anything I should change in this PR?

**Comment by tomchentw** [comment]
@Simek I'd like to clean up the commit messages a little bit, did you check "Allow Edit from collaborators"?

File: src/macros/GroundOverlay.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+/**
+ * @url https://developers.google.com/maps/documentation/javascript/reference#GroundOverlay
+ */
+export class GroundOverlay extends React.PureComponent {
+  static propTypes = {
+    __jscodeshiftPlaceholder__: null,
+    /**
+     * @type string
+     */
+    url: PropTypes.string.isRequired,
```
Comment (tomchentw) [inline src/macros/GroundOverlay.jsx]: Don't manually include these props, they're meant to be generated automatically. Just `jscodeshiftPlaceholder`

File: src/macros/GroundOverlay.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    /**
+     * @see https://developers.google.com/maps/documentation/javascript/reference#GroundOverlay
+     */
+    bounds: PropTypes.object.isRequired,
```
Comment (tomchentw) [inline src/macros/GroundOverlay.jsx]: Don't manually include these props, they're meant to be generated automatically. Just `jscodeshiftPlaceholder`

**Comment by Simek** [comment]
@tomchentw removing these props from macro and rebuilding results in missing props in documentation, but layer works as intended. How can I assure that these props will appear in docs?

**Comment by tomchentw** [comment]
@Simek My apologies, I realized that both url and bounds are for the constructor  args: `GroundOverlay(url:string, bounds:LatLngBounds|LatLngBoundsLiteral, opts?:GroundOverlayOptions)`. What you're currently doing is right, thanks!

**Comment by Simek** [comment]
@tomchentw I've added you to my fork collaborators, this should give you permission to edit PR.

**Comment by tomchentw** [comment]
Wow this is definitely awesome! Great work @Simek !
screenshot-2017-9-21 groundoverlay react google maps style guide


I reworded the commits and forced pushed to the branch. Let me know if it looks okay.

**Comment by Simek** [comment]
Seems good for me, thank you @tomchentw for your support and guidance.

**Comment by tomchentw** [comment]
Released [v8.3.0](https://github.com/tomchentw/react-google-maps/releases/tag/v8.3.0)

**Comment by DaddyWarbucks** [comment]
Thanks for the work on this!

I noticed the component does not update when new `bounds` prop is passed.  Is this by design or an oversight?

I am using an editable `<Rectangle>` component in conjunction with the `<GroundOverlay>` to allow the user to "place" the `<GroundOverlay>` by updating its props with each rectangle 'onBoundsChanged'. But, the `<GroundOverlay>` does not re-render because it  is only "drawn" via the google maps api within it constructor, not within the `componentWillReceiveProps`.

**Comment by tomchentw** [comment]
> I noticed the component does not update when new bounds prop is passed. Is this by design or an oversight?

@DaddyWarbucks if you look at the docs of Google Maps API:
https://developers.google.com/maps/documentation/javascript/reference#GroundOverlay

You'll find out there's no `setBounds` method defined on the GroundOverlay class. That's why `react-google-maps` cannot update the bounds for you, because we simply can't.

>  forcing a component remount by adding a "unique" key each render.

That's should temporarily solve the problem for sure.

@Simek any good suggestions?

**Comment by Simek** [comment]
@DaddyWarbucks Unfortunately `url` and `bounds` are constructor parameters for GroundOverlay layer and we lacking setters in Maps API. At this moment I don't have better idea how to achieve re-render without adding `key` parameter as you mentioned. But I'll try to look at this later this week.

**Comment by DaddyWarbucks** [comment]
That all makes perfect sense. Perhaps we could put the GroundOverlay construction in the render method? Since that is the actual "dom manipulation" that is a result of a state/prop change, it seems like a natural place for it. I feel like you guys may have already thought of that and put it in the constructor for good reason?

Thanks again for the work on this. I had used this lib in another project a while back and I was glad to see how well it is being maintained!

**Comment by tomchentw** [comment]
> put it in the constructor for good reason?

Just to make it align with other components in this library.

>  Perhaps we could put the GroundOverlay construction in the render method?

Maybe putting into some other lifecycle methods would be more correct. `render` should be *pure*.

**Comment by tomchentw** [comment]
PRs are welcomed!

**Comment by tomchentw** [comment]
Released [v9.0.1](https://github.com/tomchentw/react-google-maps/releases/tag/v9.0.1)

**Pull request merged**